numpy/_core/tests/test_stringdtype.py

```diff
@@ -765,6 +765,12 @@ def test_multiply_reduce():
     assert res == val * np.prod(repeats)
 
 
+def test_multiply_two_string_raises():
+    arr = np.array(["hello", "world"])
+    with pytest.raises(TypeError):
```

Comment:
Does this still raise the `ambiguous promotor` one? If so, that seems a problem... Though I guess this is a regression test? I'd actually add a `match=` for the correct expected message then.

Reply:
Well, I think you can go both ways:  Say that it is a hard error if someone does this, since everyone should make sure to never hit the code path!
But from a user perspective, it is also a promotion error like any other (even if the reason is probably a bug).

So I am fine with a generic promotion error but the text containing the details.  I agree that is just a band-aid for hitting the path!  String multiply should fail gracefully without hitting it.